### ISSUE TITLE
counsel-rg occur jumps to where counsel-rg was initiated

### ISSUE THREAD

**Issue by ItsHoff** [abo-abo/swiper#2207]
Trying to jump to an occurence from a counsel-rg occur buffer jumps to the location where counsel-rg was initiated from instead of the occurence. Tested on windows.

**Comment by abo-abo** [comment]
Does calling `next-error` work? Can you test on Linux?

**Comment by ItsHoff** [comment]
`next-error` works correctly and as well as `compile-goto-error`, so it's `ivy-occur-press-and-switch` and `ivy-occur-click` that jump to wrong location. 

I tested this on WSL and there `ivy-occur-press-and-switch` jumps to the right location.

**Comment by abo-abo** [comment]
OK, so the bug is confirmed to be Windows-only. I'm not really a Windows user, but I tried to reproduce it with `counsel-ag` (could not find ripgrep on Cygwin), and there was no problem.

If you can look into fixing the issue yourself, PRs are welcome.

**Comment by mookid** [comment]
@ItsHoff do you jump by pressing RET in the occur buffer?
I can't reproduce it.

**Comment by ItsHoff** [comment]
@mookid Yes, I use RET and the default binding `ivy-occur-press-and-switch` does not work for me. However, when `compilation-button-map` is activated RET is overridden by `compile-goto-error`, which does work. So, can you check what function RET is actually calling for you?

**Comment by abo-abo** [comment]
Any more info on this?

**Comment by ItsHoff** [comment]
Well, I just updated ivy and my workaround broke presumably due to https://github.com/abo-abo/swiper/commit/c0aa563b75cbd7b47cb6859cc28dc093e90cb3dd. So, I decided to take another look at this issue. I think the issue is caused by the path separator: ripgrep uses '\\' by default on Windows. When I tried calling ripgrep with "rg --path-separator /" both `ivy-occur-press`  and `next-error` worked as expected.

**Comment by abo-abo** [comment]
Let's make --path-separator flag the default. Can you make a PR?

**Comment by ItsHoff** [comment]
There you go.
